osquery/logger/plugins/http_logger.cpp

```diff
@@ -0,0 +1,195 @@
+      s);
+  return stat;
+}
+
+Status HTTPLoggerPlugin::logStatus(const std::vector<StatusLogLine>& log) {
+  for (const auto& item : log) {
+    boost::property_tree::ptree new_log;
+    new_log.put("action", "status_log");
+    new_log.put("severity", (google::LogSeverity)item.severity);
+    new_log.put("content", item.filename + " " + std::to_string(item.line) + " " + item.message);
+    std::stringstream ss;
+    write_json(ss, new_log);
+    auto stat = DBHandle::getInstance()->Put(
+        kLogs, std::to_string(osquery::getUnixTime()) + "_" +
+                   std::to_string(++log_num),
+        ss.str());
+    if(!stat.ok()){
+      std::cout << stat.getMessage() << "\n";
+      return stat;
+    }
+  }
+  return Status(0, "OK");
+}
+
+Status HTTPLoggerPlugin::init(const std::string& name,
+                                    const std::vector<StatusLogLine>& log) {
+  // Stop the internal glog facilities.
+  google::ShutdownGoogleLogging();
+
+  PluginResponse resp;
+  Status s = Registry::call("enrollment", "get_key", {{"enroll", "0"}}, resp);
+  // If the call was successful and returned an enrollment key
+  if (!s.ok() || resp.size() == 0 || resp[0]["key"].length() == 0) {
+    // We could not get an enrollment key, so we can send logs, best we can do
+    // is log to stderr
+    FLAGS_logtostderr = true;
+  }
+  // Restart the glog facilities using the name init was provided.
+  google::InitGoogleLogging(name.c_str());
+  Status stat = logStatus(log);
+  if(!stat.ok()){
+    return stat;
+  }
+  return Status(0, "OK");
+}
+
+class LogForwarder : public InternalRunnable{
+  LogForwarder(){}
+
+  ~LogForwarder(){}
+
+  /// Anything that needs to happen to verify that the LogForwarder can run
+  Status initVerify() {
+    if (FLAGS_max_data_size == 0) {
+      // VLOG(1) << "Could not start log forwarding loop, max data size not
+      // set";
+      return Status(1, "NOK");
+    }
+    if (FLAGS_log_force_send_time == 0) {
+      // VLOG(1) << "Could not start log forwarding loop, loop amount not set";
+      return Status(1, "NOK");
+    }
+    if (FLAGS_log_check_interval == 0) {
+      // VLOG(1) << "Could not start log forwarding loop, sleep amount not set";
+      return Status(1, "NOK");
+    }
+    if (FLAGS_log_force_send_threshold == 0) {
+      // VLOG(1) << "Could not start log forwarding loop, loop threshold not
+      // set";
+      return Status(1, "NOK");
+    }
+    if (!DBHandle::getInstance()->checkDB()) {
+      // VLOG(1) << "Could not start log forwarding loop, database checkDB
+      // failed";
+      return Status(1, "NOK");
+    }
+    return Status(0, "OK");
+  }
+
+  void enter() {
+    // Verify configuration is good
+    if (!initVerify().ok()) {
+      // We couldn't setup remote logging, so try to use filesystem
+      return;
+    }
+    std::vector<std::string> logs;
+
+    while (true) {
+      for (int i = 0; i < FLAGS_log_force_send_time / FLAGS_log_check_interval;
```

Comment:
Can you drop this entire block into another method. The right-heavy context is unsettling. Organizing your runloop into a method is easier to read. ;)